x-pack/legacy/plugins/apm/server/lib/service_map/get_next_transaction_samples.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { uniq } from 'lodash';
+import { ESClient } from '../helpers/es_client';
+import { Transaction } from '../../../typings/es_schemas/ui/Transaction';
+import {
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  DESTINATION_ADDRESS,
+  TRACE_ID,
+  SPAN_DURATION,
+  SPAN_TYPE,
+  SPAN_SUBTYPE,
+  TIMESTAMP
+} from '../../../common/elasticsearch_fieldnames';
+
+export async function getNextTransactionSamples({
+  targetApmIndices,
+  startTimeInterval,
+  afterKey,
+  esClient
+}: {
+  targetApmIndices: string[];
+  startTimeInterval: string | number;
+  afterKey?: object;
+  esClient: ESClient;
+}) {
+  const params = {
+    index: targetApmIndices,
```

Comment:
It's not immediately possible to see which indices are being queried. Everywhere else `apmIndices` is passed in. Can we do the same here?
```suggestion
    index: [apmIndices['apm_oss.transactionIndices'], apmIndices['apm_oss.spanIndices']]
```